_source/_posts/2022-05-15-jhipster-ionic-spring-boot.adoc

```diff
@@ -0,0 +1,481 @@
+description: "Build an Ionic frontend for your Spring Boot backend, all using the power of JHipster!"
+tags: [ionic, spring-boot, spring-security, jhipster]
+tweets:
+- "Create an Ionic app with a Spring Boot backend in minutes with the new Ionic blueprint for @jhipster!"
+- "Introducing JHipster Ionic: 'Wow! Did you really just build that?!'"
+- "Ionic is awesome. Learn how you to create apps quickly using this powerful framework. 💪"
+image:
+type: conversion
+github: https://github.com/oktadev/okta-jhipster-ionic-example
+---
+:page-liquid:
+:toc: macro
+:experimental:
+:commandkey: &#8984;
+
+[comment]
+--
+todo: research and figure out best title.
+Other options:
+
+- Integrate Ionic and Spring Boot with OAuth/OIDC
+- Integrate Ionic and Spring Boot
+- Ionic for JHipster
+--
+
+Ionic allows you to create mobile apps with web technologies that look and act like native apps.
+Because they're built with web technologies (HTML, JavaScript, and CSS), you can also deploy your Ionic apps as webapps.
+Or, even better, as progressive web apps (PWAs) that work offline.
+
+// image below borrowed from https://capacitorjs.com/
+image::{% asset_path 'blog/jhipster-ionic-blueprint/capacitor.png' %}[alt=Capacitor diagram,width=300,role="pull-right m-up-20",link={% asset_path 'blog/jhipster-ionic-blueprint/capacitor.png' %}]
+
+Ionic provides support for the big three web frameworks: Angular, React, and Vue.
+Once you've written your app with one of these, you can deploy it to a simulator or device with https://capacitorjs.com/[Capacitor].
+Capacitor (pictured on the right as the blue layer) provides the runtime for your app to communicate with the native operating system and vise versa.
+
+Ionic's main competitors are native apps built with Swift or Objective-C (for iOS) and Java or Kotlin (for Android).
+Ionic also competes with React Native, which uses web technologies and translates them to native components.
+
+The Ionic blog has a recent post that https://ionicframework.com/blog/ionic-vs-react-native-performance-comparison/[does a performance comparison between Ionic and React Native].
+TL;DR: Both options will give you a high performance app with a truly native look and feel.
+
+**Prerequisites:**
+
+- https://nodejs.org[Node 16]
+- https://sdkman.io/[Java 11]
+- https://docs.docker.com/desktop/#download-and-install[Docker Desktop]
+
+toc::[]
+
+== ✨ Introducing the JHipster Ionic blueprint!
+
+The JHipster project has provided support for generating an Ionic app using a `generator-jhipster-ionic` module for the past several years.
+As the primary maintainer of this module, I'm proud to announce that this module has been re-written as a blueprint and it's _much_ easier to understand now.
+The previous module relied on the Ionic CLI, the base Angular https://github.com/ionic-team/starters[starter], a https://github.com/oktadev/ionic-jhipster-starter[Ionic JHipster starter], and custom code to glue it all together.
```

Comment:
```suggestion
The previous module relied on the Ionic CLI, the base Angular https://github.com/ionic-team/starters[starter], the https://github.com/oktadev/ionic-jhipster-starter[Ionic JHipster starter], and custom code to glue it all together.
```
Or should it be "an Ionic JHipster starter" ? Your call.

Reply:
I think "the" is OK.